_posts/2021-12-3-arrow-duckdb.md

```diff
@@ -0,0 +1,383 @@
+
+This post is a collaboration with and cross-posted on [the DuckDB blog](https://duckdb.org/2021/12/03/duck-arrow.html).
+
+Part of [Apache Arrow](https://arrow.apache.org) is an in-memory data format optimized for analytical libraries. Like Pandas and R Dataframes, it uses a columnar data model. But the Arrow project contains more than just the format: The Arrow C++ library, which is accessible in Python, R, and Ruby via bindings, has additional features that allow you to compute efficiently on datasets. These additional features are on top of the implementation of the in-memory format described above. The datasets may span multiple files in Parquet, CSV, or other formats, and files may even be on remote or cloud storage like HDFS or Amazon S3. The Arrow C++ query engine supports the streaming of query results, has an efficient implementation of complex data types (e.g., Lists, Structs, Maps), and can perform important scan optimizations like Projection and Filter Pushdown.
+
+[DuckDB](https://www.duckdb.org) is a new analytical data management system that is designed to run complex SQL queries within other processes. DuckDB has bindings for R and Python, among others. DuckDB can query Arrow datasets directly and stream query results back to Arrow. This integration allows users to query Arrow data using DuckDB's SQL Interface and API, while taking advantage of DuckDB's parallel vectorized execution engine, without requiring any extra data copying.
+
+There are three main aspects that make this integration unique:
+
+1. **Larger Than Memory Analysis:** Since both libraries support streaming query results, we are capable of executing on data without fully loading it from disk. Instead, we can execute one batch at a time. This allows us to execute queries on data that is bigger than memory.
+2. **Complex Data Types:** DuckDB can efficiently process complex data types that can be stored in Arrow vectors, including arbitrarily nested structs, lists, and maps.
+3. **Advanced Optimizer:** DuckDB's state-of-the-art optimizer can push down filters and projections directly into Arrow scans. As a result, only relevant columns and partitions will be read, allowing the system to e.g., take advantage of partition elimination in Parquet files. This significantly accelerates query execution.
+
+For those that are just interested in benchmarks, you can jump ahead [benchmark section below](#Benchmark Comparison).
+
+## Quick Tour
+Before diving into the details of the integration, in this section we provide a quick motivating example of how powerful and simple to use is the DuckDB-Arrow integration. With a few lines of code, you can already start querying Arrow datasets. Say you want to analyze the infamous [NYC Taxi Dataset](https://www1.nyc.gov/site/tlc/about/tlc-trip-record-data.page) and figure out if groups tip more or less than single riders.
```

Comment:
One thing that I'm not sure about is that we talk about "(Arrow) datasets" as a thing but never really define it, and I wonder if people might walk away confused, since there's "an Arrow Dataset object", a "dataset of files", and "a dataset of Arrow (IPC) files". This also occurs below on L180 (where we also refer to Datasets as a proper noun).

Reply:
We could link to https://arrow.apache.org/docs/cpp/dataset.html, (and/or the python and R versions of those docs)